frontend/simple/js/state.js

```diff
@@ -5,22 +5,123 @@
 
 import Vue from 'vue'
 import Vuex from 'vuex'
-
-const S = require('string')
+import EventLog from './event-log'
+import {EVENT_TYPE} from '../../../shared/constants'
+const {SUCCESS} = EVENT_TYPE
 
 Vue.use(Vuex)
 
 // TODO: save only relevant state to localforage
 const state = {
+  currentGroupLog: null,
+  availableGroups: [],
   // TODO: this should be managed by Keychain, not here
-  loggedIn: false
+  loggedInUser: null,
+  offset: [],
+  socket: null
 }
 
 // Mutations must be synchronous! Never call these directly!
 // http://vuex.vuejs.org/en/mutations.html
 const mutations = {
-  LOGIN (state) { state.loggedIn = true },
-  LOGOUT (state) { state.loggedIn = false }
+  login (state, user) { state.loggedInUser = user },
+  logout (state) { state.loggedInUser = null },
+  updateCurrentGroupLog (state, log) {
+    if (Array.isArray(log.offset)) {
+      state.currentGroupLog = log.log
+      state.offset = log.offset
+    } else {
+      state.currentGroupLog = log
+    }
+  },
+  updateAvailableGroups (state, available) {
+    state.availableGroups = available
+  },
+  updateSocket (state, socket) {
+    state.socket = socket
+  }
+}
+
+export const actions = {
+  async createGroup ({commit, state}, group) {
+    let log = await EventLog.createEventLogFromGroup(group)
+    let available = state.availableGroups
+    if (state.socket) {
+      await joinRoom(state.socket, group)
+    }
+    commit('updateCurrentGroupLog', log)
+    available.push(log.currentLogPosition)
```

Comment:
What I meant was you are directly writing to the state on line 54 when you push into it. That is not OK per Vuex docs.